### PR TITLE
Add `read_vcf` convenience function

### PR DESCRIPTION
**Pull request by tomwhite**
- [x] Partial fix for #1004
- [x] Tests added
- [x] User visible changes (including notable bug fixes) are documented in `changelog.rst`
- [x] New functions are listed in `api.rst`

Uses the new function from #1046.

### CODE REVIEW

**Comment by tomwhite** [comment]
I just realised that this conflicts with #1044, so that one should go in first.

File: sgkit/io/vcf/vcf_reader.py [hunk 1]
```diff
@@ -990,6 +991,139 @@ def vcf_to_zarr(
...
+
+    """
+
+    # Need to retain zarr file backing the returned dataset
+    with temporary_directory(
+        prefix="read_vcf_",
+        suffix=".zarr",
+        dir=tempdir,
+        storage_options=tempdir_storage_options,
+        retain_temp_files=True,
```
Comment (benjeffery) [inline sgkit/io/vcf/vcf_reader.py:1102]: Glad this has come in useful!

**Review by benjeffery — APPROVED** [review]
LGTM, just needs changelog.

**Comment by tomwhite** [comment]
Updated the changelog. @benjeffery do want me to wait until #1044 is done before this one goes in?

**Comment by benjeffery** [comment]
@tomwhite #1044 is done, so if you want to merge that first and then add the new option here then that probably makes sense? Happy to do it the other way round though if you like.

**Pull request merged**